gordo_components/builder/mlflow_utils.py

```diff
@@ -474,20 +477,9 @@ def log_metadata(mlflow_client, run_id, metadata: dict):
 
     # Send configs as JSON artifacts
     try:
-        with tempfile.TemporaryDirectory() as tmp_dir:
-            for name, obj in [
-                ("model-config", metadata["model"]),
-                ("dataset-config", metadata["dataset"]),
-                (
-                    "user-config",
-                    metadata.get("user-defined") or metadata.get("metadata"),
-                ),
-            ]:
-                fp = os.path.join(tmp_dir, f"{name}.json")
-                json.dump(obj, open(fp, "w"), cls=DatetimeEncoder)
-            mlflow_client.log_artifacts(run_id, local_dir=tmp_dir)
-    # Log error without exiting the build
-    # FIX: when a solution is found when logging artifacts wit azureml lib,
-    # this can raise an error
+        with tempfile.TemporaryDirectory(dir="./") as tmp_dir:
+            fp = os.path.join(tmp_dir, f"metadata.json")
+            json.dump(metadata, open(fp, "w"), cls=DatetimeEncoder)
+            mlflow_client.log_artifacts(run_id, tmp_dir)
     except Exception as e:
-        logger.error(e)
+        raise MlflowLoggingError(e)
```

Comment:
Why are we doing this? Seems it will make things harder to debug, in that if say it errors serializing with a TypeError, we don't get a `TypeError` at all. Is it not better to simply remove the try catch?

Reply:
I see a bit of a trade off there. :thinking: Maybe a simple comment to mention why remapping the error is necessary, Otherwise it just looks strange. IMO

Reply:
Roger. Commented :heavy_check_mark: